front/src/modules/simulationResult/components/ChartHelpers/drawCurve.ts

```diff
@@ -23,22 +23,41 @@ const drawCurve = <
   const xAxis = getAxis(keyValues, 'x', rotate);
   const yAxis = getAxis(keyValues, 'y', rotate);
 
+  const isHeight = yAxis === 'height';
+
   type Key = keyof T;
 
-  drawZone
-    .append('path')
-    .attr('class', `line zoomable ${classes}`)
-    .datum(dataSimulation)
-    .attr('fill', 'none')
-    .attr('stroke-width', 1)
-    .attr(
-      'd',
-      d3
-        .line<T>()
-        .x((d) => chart.x(d[xAxis as Key] as number))
-        .y((d) => chart.y(d[yAxis as Key] as number))
-        .curve(d3[interpolation as keyof d3.CurveFactory])
-    );
+  if (isHeight && chart.y2) {
+    drawZone
+      .append('path')
+      .attr('class', `line zoomable additional-y ${classes}`)
+      .datum(dataSimulation)
+      .attr('fill', 'none')
+      .attr('stroke-width', 1)
+      .attr(
+        'd',
+        d3
+          .line<T>()
+          .x((d) => chart.x(d[xAxis as keyof T] as number))
+          .y((d) => chart.y2!(d[yAxis as keyof T] as number))
+          .curve(d3[interpolation as keyof d3.CurveFactory])
+      );
+  } else {
+    drawZone
+      .append('path')
+      .attr('class', `line zoomable ${classes}`)
+      .datum(dataSimulation)
+      .attr('fill', 'none')
+      .attr('stroke-width', 1)
+      .attr(
+        'd',
+        d3
+          .line<T>()
+          .x((d) => chart.x(d[xAxis as Key] as number))
+          .y((d) => chart.y(d[yAxis as Key] as number))
+          .curve(d3[interpolation as keyof d3.CurveFactory])
+      );
+  }
```

Comment:
```suggestion
  drawZone
    .append('path')
    .attr('class', `line zoomable ${onY2Axis && 'additional-y'} ${classes}`)
    .datum(dataSimulation)
    .attr('fill', 'none')
    .attr('stroke-width', 1)
    .attr(
      'd',
      d3
        .line<T>()
        .x((d) => chart.x(d[xAxis as keyof T] as number))
        .y((d) =>
          onY2Axis && chart.y2
            ? chart.y2(d[yAxis as keyof T] as number)
            : chart.y(d[yAxis as Key] as number)
        )
        .curve(d3[interpolation as keyof d3.CurveFactory])
    );
```

It's not very pretty because we have to test twice that chart.y2 exists, as typescript does not detect it alone, but at least it avoids code duplication and you can clearly see what changes between the 2 cases. You can also remove the second test chart.y2 and add a ! if you prefer (`chart.y2!(...)` line 43)